docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+chosen chunk of scrambled transactions, they send revealed encryption keys to
+the leader to be propagated via turbine likewise the block shreds. Also
+sig-verify is done at the end of descrambling.
+
+After all transactions are descrambled from the cooperative efforts, the block
+can be finally scheduled and executed deterministically.
+
+Lastly, validators votes on the reproduced bank hash after replaying the
+processing likewise. Then, rpc can notify optimistic confirmation signal to
+clients.
+
+### Downsides
+
+- Increased latency
+- Some computational cost on clients
+- Reliance of local wall clock time
+- Pretty invasive system change
+
+
+### Assumptions of other subsystem's improvements
+
+- bankless leader
+- randomized leader schedule (or stop 4-slot grouping at the very least)
+- on-chain mempool with bidding
+- quic adaption for tpu (for ipv6-mtu-exeeding transaction packet and ddos prevention)
+
+### Scrambling details
+
+As said above, scrambling refers to a specially-arranged encryption scheme where
+anyone can decrypt (= reveal) the transaction details after a non-parallelizable
+computation which takes certain amount of time.
+
+On top of it, scrambling also has a nice property where the revelation can
+easily be verified by anyone with some kind of message authentication tag. This
+means rpc node doesn't need GPU.
+
+All of fields in a transaction are scrambled with randomized padding, including
+fee payer to avoid any kind of fingerprinting while allowing garbage-based
+spamming with amplification risk.
+
+Also, scrambled transaction expires very quickly based on wall clock time, less
+than the targeted duration of difficulty to protect from processing stalls by
+malicious leaders for MEV.
+
+So, scrambled transaction binary format will be like this
+
+```
+[scrambled transaction marker]
+[seed to recursive sha256]
+[recursion count]
+aes256-ed..
+  [aes256 key used to scramble] # this works as authentication tag
+  [expiration in unix time]
+  [normal transaction details...] # fee payer sig cover the scrambling key to avoid unauthorized scrambling
+```
+
+So, this assumes quic to bypass the transaction udp packet size limit.
+
+The number of sha256 recursion is chosen so that anyone can't reveal the
+transaction details, until the cluster optimistically commits the ordering of
+transactions in a block with very high certainty. So, this assumes the recursive
+sha256 is exhibits rather uniform throughput across various hardware (cpu/gpu)
+on the market.
+
+For instance, difficulty can be adjusted 6 9s of availability of MEV-immunity
+status based on standard deviation of past conf time. or hard-coded initially.
+
+Scrambling is optional. While execution of not-scrambled transaction is stalled
+still, infallible and/or MEV-free simple transactions can opt not to be
+scrambled because scrambling might not be desirable for all of situations (HD
+wallets and program deploy transactions).
```

Comment:
I wonder if the complexity of choosing between MEV-protected and MEV-unprotected transactions can be abstracted away from the end-user at the program (or maybe wallet) level, because they shouldn't know anything about MEV really.

Reply:
yep, i'm thinking wallet should and can well handle the choice.